src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (object.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) object;
+            return width.equals(rectangle.width)
+                    && length.equals(rectangle.length)
+                    && color.equals(rectangle.color);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int generatedHashCode = 9;
+        generatedHashCode = generatedHashCode * 19 + width;
+        generatedHashCode = generatedHashCode * 19 + length;
+        generatedHashCode = generatedHashCode * 19 + color.hashCode();
```

Comment:
what happens, if color == null ?

Reply:
then we will add just 0 as hashCode(null) = 0, is that prohibited?

Reply:
you sure that there won't be nullPointerException if color == null? 
it's not hashCode(null) it's null.hashCode()

Reply:
null check added, I think that may help